### DISCUSSION TITLE
Specific class for specific composite waveform?

### DISCUSSION THREAD

**Discussion by LaurentAjdnik** [pasqal-io/Pulser#212] [Ideas] [answered]
@HGSilveri recently added the `InterpolatedWaveform` class and we now have an extensive set of elementary waveforms to concatenate into composite waveforms.

Still, I wonder if one these composite waveforms is a **common enough** use case to deserve its own class:
* Constant value for some duration
* Ramp up (or down) for some duration
* Constant value for some duration
* Ramp down (or up) for some duration
* Constant value (same as first one?) for some duration

This would require a single call instead of creating many waveforms and "composing" them.

Plus, by setting any/many of the durations to 0, this versatile generator could produce 27 different waveforms (+ 5 being existing hence meaningless shapes: 3 `ConstantWaveform` and 2 `RampWaveform`).

It could be a subclass of `CompositeWaveform` or a class by itself (calling `CompositeWaveform` anyway).

What do you think?

**Comment by HGSilveri** [comment, marked as answer]
It's a cool idea, though I'm unsure of how needed it is. If needed, the user has all the pieces to write a simple function that does this automatically. I would rather not add something whose need has not presented itself yet. That being said, have you encountered a need for something like this?

**Reply by LaurentAjdnik** [reply]
> It's a cool idea, though I'm unsure of how needed it is. If needed, the user has all the pieces to write a simple function that does this automatically. I would rather not add something whose need has not presented itself yet. That being said, have you encountered a need for something like this?

Nope, I was just wondering whether you or anybody else had. We'll explore this idea further some day if a user requests such a feature. You can set it as "Answered".

**Reply by HGSilveri** [reply]
Alright, thank you for the suggestion!
